src/contract/helpers.js

```diff
@@ -0,0 +1,158 @@
+  const pools = await StakingViewer.getAllPools();
+  return pools.reduce((acc, pool) => {
+    acc[pool.poolId.toString()] = pool;
+    return acc;
+  }, {});
+}
+
+async function fetchStakingPoolDataById(id) {
+  const StakingViewer = new ethers.Contract(CONTRACTS_ADDRESSES.StakingViewer, StakingViewerAbi, provider);
+
+  return StakingViewer.getPool(id);
+}
+
+/* ================== PRODUCTS ================================== */
+
+async function fetchProductDataById(id, globalCapacityRatio) {
+  const StakingViewer = new ethers.Contract(CONTRACTS_ADDRESSES.StakingViewer, StakingViewerAbi, provider);
+  const Cover = new ethers.Contract(CONTRACTS_ADDRESSES.Cover, CoverAbi, provider);
+
+  if (!globalCapacityRatio) {
+    globalCapacityRatio = await Cover.globalCapacityRatio();
+  }
+  const pools = await StakingViewer.getProductPools(id);
+  const { capacityReductionRatio } = await Cover.products(id);
+  const latestBlockNumber = await provider.getBlockNumber();
+  const productData = {};
+
+  for (const pool of pools) {
+    productData[pool.poolId] = await fetchProductDataForPool(id, pool.poolId.toNumber(), {
+      globalCapacityRatio,
+      capacityReductionRatio,
+      latestBlockNumber,
+    });
+  }
+  return productData;
+}
+
+async function fetchAllProductsData() {
+  const Cover = new ethers.Contract(CONTRACTS_ADDRESSES.Cover, CoverAbi, provider);
+
+  const globalCapacityRatio = await Cover.globalCapacityRatio();
+  const products = await Cover.getProducts();
+  const productsData = {};
+
+  for (const productId in products) {
+    productsData[productId] = await fetchProductDataById(productId, globalCapacityRatio);
+  }
+  return productsData;
+}
+
+async function fetchAllProductDataForPool(poolId) {
+  const StakingViewer = new ethers.Contract(CONTRACTS_ADDRESSES.StakingViewer, StakingViewerAbi, provider);
+
+  const products = StakingViewer.getPoolProducts(poolId);
+  const stakingPoolProducts = [];
+  for (const { productId } of products) {
+    const data = await fetchProductDataForPool(productId, poolId);
+    stakingPoolProducts.push({ ...data, productId });
+  }
+  return stakingPoolProducts;
+}
+
+async function fetchProductDataForPool(
+  productId,
+  poolId,
+  { latestBlockNumber, capacityReductionRatio, globalCapacityRatio } = {},
+) {
+  const address = calculateAddress(poolId);
+  const Cover = new ethers.Contract(CONTRACTS_ADDRESSES.Cover, CoverAbi, provider);
+  const StakingPool = new ethers.Contract(address, StakingPoolAbi, provider);
+  const StakingProducts = new ethers.Contract(CONTRACTS_ADDRESSES.StakingProducts, StakingProductAbi, provider);
+
+  if (!latestBlockNumber) {
+    latestBlockNumber = await provider.getBlockNumber();
+  }
+  if (!capacityReductionRatio) {
+    const product = await Cover.products(productId);
+    capacityReductionRatio = product.capacityReductionRatio;
+  }
+  if (!globalCapacityRatio) {
+    globalCapacityRatio = await Cover.globalCapacityRatio();
```

Comment:
You're fetching this in a few different places. Capacity factors need to be in the store.